### PR TITLE
Update P/Invoke sample to find MSVC compiler as part of build

### PR DESCRIPTION
**Pull request by elinor-fung**
## Summary

- Update P/Invoke sample to find MSVC compiler as part of build
- Add metadata to readme file for indexing in samples browser

cc @vitek-karas @AaronRobinsonMSFT @jkoritzinsky

### CODE REVIEW

**Comment by elinor-fung** [comment]
@adegeo This is using the same method as https://github.com/dotnet/samples/pull/3744#issuecomment-662703028 for finding the MSVC compiler. The CI looks to be happy now.

File: core/interop/pinvoke/marshaling/src/MarshalingSample/MarshalingSample.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>netcoreapp3.0</TargetFramework>
     <OutputType>exe</OutputType>
+    <UseAppHost>true</UseAppHost>
```
Comment (vitek-karas) [inline core/interop/pinvoke/marshaling/src/MarshalingSample/MarshalingSample.csproj:6]: Just curious why is this necessary - I thought that "exe" projects always have `UseAppHost` set to true (for 3.0 and above).

File: core/interop/pinvoke/marshaling/readme.md [hunk 2]
```diff
@@ -1,26 +1,35 @@
...
+# .NET Core P/Invoke Marshaling Sample
 
-Prerequisites
-------------
+This project demonstrates different ways to marshal arguments to native functions when using P/Invokes. Documentation can be found here for [P/Invokes](https://docs.microsoft.com/dotnet/standard/native-interop/pinvoke) and here for the [Type marshaling](https://docs.microsoft.com/dotnet/standard/native-interop/type-marshaling).
 
-* .NET Core 3.0 (at least Preview 6) - [https://dot.net](https://github.com/dotnet/core-sdk#installers-and-binaries)
+## Prerequisites
+
+* [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download) (at least Preview 6)
```
Comment (BillWagner) [inline core/interop/pinvoke/marshaling/readme.md:19]: At this point, should we recommend .NET Core SDK 3.1 (It's the LTS release, and 3.0 is already out of support).

File: core/interop/pinvoke/marshaling/src/MarshalingSample/MarshalingSample.csproj (hunk 1 above)
Comment (adegeo) [reply core/interop/pinvoke/marshaling/src/MarshalingSample/MarshalingSample.csproj:6]: Not on OSX

File: core/interop/pinvoke/marshaling/readme.md (hunk 2 above)
Comment (elinor-fung) [reply core/interop/pinvoke/marshaling/readme.md]: That makes sense to me. Updated.

File: core/interop/pinvoke/marshaling/src/MarshalingSample/MarshalingSample.csproj (hunk 1 above)
Comment (adegeo) [reply core/interop/pinvoke/marshaling/src/MarshalingSample/MarshalingSample.csproj]: err yes that's what I meant ;)

File: core/interop/pinvoke/marshaling/readme.md [hunk 3]
```diff
@@ -1,26 +1,35 @@
-.NET Core PInvoke Marshaling Sample
-================
+---
```
Comment (Youssef1313) [inline core/interop/pinvoke/marshaling/readme.md:1]: I believe it's more common to have the file name as "README.md" (README is all uppercase).

File: core/interop/pinvoke/marshaling/readme.md [hunk 4]
```diff
@@ -1,26 +1,35 @@
...
 
-* .NET Core 3.0 (at least Preview 6) - [https://dot.net](https://github.com/dotnet/core-sdk#installers-and-binaries)
+## Prerequisites
+
+* [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download)
 
 * C++ compiler
   * Windows: `cl.exe`
+    * See [installation instructions](https://docs.microsoft.com/cpp/build/building-on-the-command-line#download-and-install-the-tools).
   * Linux/OSX: `g++`
```
Comment (Youssef1313) [inline core/interop/pinvoke/marshaling/readme.md:24]:
```suggestion
  * Linux/macOS: `g++`
```

Related to https://github.com/dotnet/docs/pull/3324#issuecomment-335331415

File: core/interop/pinvoke/marshaling/readme.md [hunk 5]
```diff
@@ -1,26 +1,35 @@
...
 
-Build and Run
--------------
+## Build and Run
 
 1) In order to build and run, all prerequisites must be installed. The following are also required:
 
-    * The C++ compiler (`cl.exe` or `g++`) must be on the path.
-      * On Windows, a [Developer Command Prompt for Visual Studio](https://docs.microsoft.com/cpp/build/building-on-the-command-line#developer_command_prompt_shortcuts) should be used.
+    * On Linux/OSX, the C++ compiler (`g++`) must be on the path.
```
Comment (Youssef1313) [inline core/interop/pinvoke/marshaling/readme.md:30]:
```suggestion
    * On Linux/macOS, the C++ compiler (`g++`) must be on the path.
```

**Comment by elinor-fung** [comment]
@BillWagner I changed the recommended version to 3.1 instead of 3.0. Any other feedback on this?
